src/mds/balancers/greedyspill.lua

```diff
@@ -1,48 +1,45 @@
-metrics = {"auth.meta_load", "all.meta_load", "req_rate", "queue_len", "cpu_load_avg"}
+local metrics = {"auth.meta_load", "all.meta_load", "req_rate", "queue_len", "cpu_load_avg"}
 
 -- Metric for balancing is the workload; also dumps metrics
-function mds_load()
-  for i=0, #mds do
-    s = "MDS"..i..": < "
-    for j=1, #metrics do
-      s = s..metrics[j].."="..mds[i][metrics[j]].." "
+local function mds_load()
+  for rank, mds in pairs(mds) do
+    local s = "MDS"..rank..": < "
+    for _, metric in ipairs(metrics) do
+      s = s..metric.."="..mds[metric].." "
     end
-    mds[i]["load"] = mds[i]["all.meta_load"]
-    BAL_LOG(0, s.."> load="..mds[i]["load"])
+    mds.load = mds["all.meta_load"]
+    BAL_LOG(5, s.."> load="..mds.load)
   end
 end
 
 -- Shed load when you have load and your neighbor doesn't
-function when()
+local function when()
+  if not mds[whoami+1] then return false end -- i'm the last rank
```

Comment:
Also, this makes me think that if C++ land fails to execute Lua code, we shouldn't fall back to the default CephFS balancer because it will start migrating load with that original policy -- it will look almost random to someone who was trying to use Mantle. Thoughts? I started implementing it but it messed up test cases in `/qa` so I need a little more time.